netsim/augment/nodes.py

```diff
@@ -334,6 +334,44 @@ def augment_node_device_data(n: Box, defaults: Box) -> None:
     log.error(f"Daemon configuration files for node {n} must be a dictionary")
     n.pop('_daemon_config',None)
 
+'''
+Set addresses of the main loopback interface
+'''
+def loopback_interface(n: Box, pools: Box, topology: Box) -> None:
+  n.loopback.type = 'loopback'
+  n.loopback.neighbors = []
+  n.loopback.virtual_interface = True
+  lbname = devices.get_loopback_name(n,topology)
+  if lbname:
+    n.loopback.ifname = lbname
+    n.loopback.ifindex = 0
```

Comment:
Shouldn't the loopback have ifindex=0 regardless whether it has a name?
Code elsewhere depends on each interface having an ```ifindex```

Reply:
The loopback ifindex is never checked in the transformation code (and a physical interface could have ifindex 0 on some platforms), the **ifname** and **ifindex** are there just to make templates that use **netlab_interfaces** (loopback + physical intf) work.